--- a/utils/opcontrol
+++ b/utils/opcontrol
@@ -371,7 +371,7 @@
 	echo "SESSION_DIR=$SESSION_DIR" >>$SETUP_FILE
 
 	if test "$NR_CHOSEN" != "0"; then
-		for f in `seq 0 $((NR_CHOSEN - 1))`; do
+		for f in `seq 0 $(($NR_CHOSEN - 1))`; do
 			get_event $f
 			echo "CHOSEN_EVENTS_${f}=$GOTEVENT" >>$SETUP_FILE
 		done
@@ -566,7 +566,7 @@
 	OPHELP_ARGS=
 
 	if test "$NR_CHOSEN" != 0; then
-		for f in `seq 0 $((NR_CHOSEN - 1))`; do
+		for f in `seq 0 $(($NR_CHOSEN - 1))`; do
 			get_event $f
 			if test "$GOTEVENT" != ""; then
 				verify_ibs $GOTEVENT
@@ -591,7 +591,7 @@
 		return
 	fi
 
-	for f in `seq 0 $((NR_CHOSEN - 1))`; do
+	for f in `seq 0 $(($NR_CHOSEN - 1))`; do
 		get_event $f
 		if test "$GOTEVENT" != ""; then
 			EVENT=`echo $GOTEVENT | awk -F: '{print $1}'`
@@ -1334,7 +1334,7 @@
 	verify_counters
 
 	OPROFILED_EVENTS=
-	for f in `seq 0 $((NR_CHOSEN - 1))`; do
+	for f in `seq 0 $(($NR_CHOSEN - 1))`; do
 		get_event $f
 		if test "$GOTEVENT" != ""; then
 			EVENT=`echo $GOTEVENT | awk -F: '{print $1}'`
@@ -1344,7 +1344,7 @@
 			UNIT_MASK=`echo $GOTEVENT | awk -F: '{print $3}'`
 			KERNEL=`echo $GOTEVENT | awk -F: '{print $4}'`
 			USER=`echo $GOTEVENT | awk -F: '{print $5}'`
-			CTR=`echo $HW_CTRS | awk "{print \\$$((f + 1))}"`
+			CTR=`echo $HW_CTRS | awk "{print \\$$(($f + 1))}"`
 			check_event_mapping_data
 
 			if test "$EVENT" = "SPU_CYCLES"; then
@@ -1519,7 +1519,7 @@
 	fi
 
 	if test "$NR_CHOSEN" != "0"; then
-		for f in `seq 0 $((NR_CHOSEN - 1))`; do
+		for f in `seq 0 $(($NR_CHOSEN - 1))`; do
 			get_event $f
 			echo "Event $f: $GOTEVENT"
 		done
